metric_providers/base.py

```diff
@@ -47,7 +47,17 @@ def __init__(
 
     # this is the default function that will be overridden in the children
     def check_system(self):
-        pass
+        cmd = ['pgrep', '-f', self._metric_provider_executable]
+        result = subprocess.run(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            check=False, encoding='UTF-8')
+        # returncode 1 sadly indicates program failure or no found process, so we need to go deeper
+        if result.returncode not in [0, 1]:
+            raise subprocess.CalledProcessError(result.stderr, cmd)
+        if result.stdout != '':
```

Comment:
Couldn't you work with return codes here? If you are already looking at the code beforehand? The Empty string check is also ok though IMHO

Reply:
see comment above. The restructuring makes the comment obsolte I think